This removes UFS quota enablement at boot, so it's a minor simplification
and performance improvement, but also allows removal of the quotacheck
and quotaon/off binaries from the system.

Copyright 2025 Peter Tribble

--- a/usr/src/cmd/svc/milestone/fs-local
+++ b/usr/src/cmd/svc/milestone/fs-local
@@ -49,42 +49,6 @@ elif [ $rc -ne 0 ]; then
 fi
 rm -f /var/run/fs-local
 
-#
-# If there are non-global UFS filesystems with quotas, check and enable them.
-#
-
-# vlist is the non-global filesystems in vfstab requesting quotas
-vlist=`/usr/bin/nawk '$1 !~ /^(#|-)/ && $4 == "ufs" {
-	if (match($7, "(^|,)(quota|rq)(,|$)") != 0 &&
-	    match($7, "(^|,)global(,|$)") == 0) print $1; }' /etc/vfstab`
-
-if [ -n "$vlist" ]; then
-	# mlist is the filesystems in mnttab that are ufs, mounted rw,
-	# and without quotas turned on
-	mlist=`/usr/sbin/mount -p | /usr/bin/nawk '$4 == "ufs" {
-		if (match($7, "(^|,)ro(,|$)") == 0) print $1; }'`
-
-	# qlist is the intersection of vlist and mlist
-	qlist=`echo "$vlist\n-\n$mlist" | \
-		/usr/bin/nawk '{if ($1 == "-") { mlist = 1; }
-			else if (mlist == 0) { vlist[$1] = 1; }
-			else if (vlist[$1]) { print $1; } }'`
-
-	#
-	# Just check and enable the non-global UFS file systems with quotas
-	# enabled. Note that "quotacheck -a" and "quotaon -a" will try
-	# to process all UFS entries with quotas rather than excluding
-	# the entries with the global option (the global entries are handled
-	# later in another script if the cluster package is installed).
-	#
-	if [ -n "$qlist" ]; then
-		echo 'Checking UFS quotas: \c'
-		/usr/sbin/quotacheck -p $qlist
-		echo 'done.'
-		/usr/sbin/quotaon $qlist
-	fi
-fi
-
 # Mount all ZFS filesystems.
 
 if [ -x /usr/sbin/zfs ]; then
